core/engine/src/native_function/mod.rs

```diff
@@ -178,6 +178,7 @@ impl NativeFunction {
     /// Certain async functions need to be desugared for them to be compatible. For example, the
     /// following won't compile:
     ///
+    /// TODO: this does not fail anymore.
```

Comment:
Uhh I'm confused by this because it is marked as `compile_fail` and the CI seems to mark it as passing...

Reply:
Ohhh, it succeeds because our imports have a missing comma lol

Reply:
Yeah, I noticed it from my IDE and didn't have time to look deeper into it. I'll revert this.